### ISSUE TITLE
Port XMPP/Jabber examples to Python 3.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8509]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#8509|
|Type|enhancement|
|Created|2016-06-27 11:24:37Z|
|Branch|https://github.com/twisted/twisted/tree/py3-example-xmpp-8509|

<details><summary>Searchable metadata</summary>

```
trac-id__8509 8509
type__enhancement enhancement
reporter__ralphm ralphm
priority__normal normal
milestone__Python_3_x Python-3.x
branch__py3_example_xmpp_8509 py3-example-xmpp-8509
branch_author__ralphm ralphm
status__closed closed
resolution__fixed fixed
component__words words
keywords__None None
time__1467026677915503 1467026677915503
changetime__1467108217692615 1467108217692615
version__None None
owner__ralphm ralphm
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm set status to `assigned`|
|-|-|

Associated GitHub PR: https://github.com/twisted/twisted/pull/232

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_3"></a>|@ralphm removed owner<br>@ralphm set status to `new`|
|-|-|

Besides porting `xmpp_client.py`, I removed the old `jabber_client.py` example as most servers don't support the ancient, pre-XMPP 1.0, [non-SASL authentication](https://xmpp.org/extensions/xep-0078.html) protocol anymore. The linked specification was deprecated in 2006 and obsoleted in 2008. Keeping it will only confuse new users.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @ralphm|
|-|-|

Looks good. Make sense. Thanks! Please merge.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub set status to `closed`|
|-|-|

In changeset a1171ec06ccf30c223674a326ab63a2c7a9e66cd
```
#!CommitTicketReference repository="" revision="a1171ec06ccf30c223674a326ab63a2c7a9e66cd"
Merge pull request #232 from twisted/py3-example-xmpp-8509

Port Jabber/XMPP examples to Python 3.

Author: ralphm
Reviewer: adiroiban
Fixes: #8509
```
